backend/src/main/kotlin/app/ehrenamtskarte/backend/cards/Argon2IdHasher.kt

```diff
@@ -69,5 +74,27 @@ class Argon2IdHasher {
             generator.generateBytes(canonicalJson.toByteArray(), result)
             return encodeWithoutSalt(result, params)
         }
+
+        fun isValidUserHash(userHash: String): Boolean {
+            val parts = userHash.trimStart('$').split("\\$".toRegex())
+            if (parts.size != 4) return false
+
+            val (algorithm, version, performance, hash) = parts
+
+            if (algorithm != DEFAULT_ALGORITHM_TYPE) return false
+            if (version != "v=$DEFAULT_VERSION") return false
+            if (hash.isEmpty()) return false
+
+            val performanceParams = performance.split(",")
+            if (performanceParams.size != 3) return false
+
+            val (memory, iterations, parallelism) = performanceParams
+
+            if (memory != "m=$DEFAULT_MEMORY") return false
+            if (iterations != "t=$DEFAULT_ITERATIONS") return false
+            if (parallelism != "p=$DEFAULT_PARALLELISM") return false
```

Comment:
If I understand your question correctly:
We don’t expect that these params will be changed unilaterally by Koblenz, and this validation is intended to prevent this from happening.

Reply:
okay, thanks for explaining :)